std/typecons.d

```diff
@@ -89,42 +96,43 @@ public:
     }
 
     /**
-    Constructor that takes an rvalue.
-    It will ensure uniqueness, as long as the rvalue
-    isn't just a view on an lvalue (e.g., a cast).
-    Typical usage:
-    ----
-    Unique!Foo f = new Foo;
+    Allows construction of a $(D Unique) for a nested class or struct.
+    Currently, references to non-static nested objects can only be created
+    using $(D new) in their original context.
+
+    To ensure uniqueness, be sure to provide the direct result of $(D new).
+
+    Example:
```

Comment:
There is a corresponding unit test - see https://github.com/mrkline/phobos/blob/better-unique/std/typecons.d#L225-L236